sync2/rangesync/rangesync.go

```diff
@@ -38,6 +38,23 @@ type RangeSetReconcilerConfig struct {
 	MessageLimit int `mapstructure:"message-limit"`
 }
 
+func (cfg *RangeSetReconcilerConfig) Validate() error {
+	var err error
+	if cfg.MaxSendRange == 0 {
+		err = errors.Join(err, errors.New("max-send-range must be positive"))
+	}
+	if cfg.ItemChunkSize == 0 {
+		err = errors.Join(err, errors.New("item-chunk-size must be positive"))
+	}
+	if cfg.SampleSize > maxSampleSize {
+		err = errors.Join(err, fmt.Errorf("bad sample-size %d (max %d)", cfg.SampleSize, maxSampleSize))
+	}
+	if cfg.MaxReconcDiff < 0 || cfg.MaxReconcDiff > 1 {
+		err = errors.Join(err, errors.New("bad max-reconc-diff"))
+	}
+	return err
+}
```

Comment:
Same here - if the goals is only to log those issues, why not log them directly and just return `true`/`false` in case the config is valid or not?

Reply:
Done